server/player/player.go

```diff
@@ -122,20 +123,27 @@ func New(name string, skin skin.Skin, pos mgl64.Vec3) *Player {
 	p.velocity.Store(mgl64.Vec3{})
 	p.immunity.Store(time.Now())
 	p.breakingPos.Store(cube.Pos{})
+	if data != nil {
+		p.load(*data)
+		p.loadInventory(data.Inventory)
+	}
 	return p
 }
 
 // NewWithSession returns a new player for a network session, so that the network session can control the
 // player.
 // A set of additional fields must be provided to initialise the player with the client's data, such as the
-// name and the skin of the player.
-func NewWithSession(name, xuid string, uuid uuid.UUID, skin skin.Skin, s *session.Session, pos mgl64.Vec3) *Player {
-	p := New(name, skin, pos)
+// name and the skin of the player. You can either pass on player data you want to load or
+// you can leave the data as nil to use default data.
+func NewWithSession(name, xuid string, uuid uuid.UUID, skin skin.Skin, s *session.Session, pos mgl64.Vec3, data *Data) *Player {
+	p := New(name, skin, pos, data)
 	p.s, p.uuid, p.xuid, p.skin = s, uuid, xuid, skin
 	p.inv, p.offHand, p.armour, p.heldSlot = s.HandleInventories()
 	p.locale, _ = language.Parse(strings.Replace(s.ClientData().LanguageCode, "_", "-", 1))
-
 	chat.Global.Subscribe(p)
+	if data != nil {
+		p.loadInventory(data.Inventory)
```

Comment:
Might want to document that this is overwritten by s.HandleInventories(). Someone might remove this on accident if it's not clear why this is here.